File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/custom/CustomGenericRequeryRepository.java
package com.coupang.springframework.data.requery.repository.custom;

import com.coupang.springframework.data.requery.core.RequeryOperations;
import com.coupang.springframework.data.requery.repository.support.RequeryEntityInformation;
import com.coupang.springframework.data.requery.repository.support.SimpleRequeryRepository;
import org.jetbrains.annotations.NotNull;

/**
 * com.coupang.springframework.data.requery.repository.custom.CustomGenericRequeryRepository
 *
 * @author debop
 * @since 18. 6. 9
 */
public class CustomGenericRequeryRepository<T, ID> extends SimpleRequeryRepository<T, ID> implements CustomGenericRepository<T, ID> {

    public CustomGenericRequeryRepository(@NotNull RequeryEntityInformation<T, ID> entityInformation,
                                          @NotNull RequeryOperations operations) {
        super(entityInformation, operations);
    }

    @Override
    public T customMethod(ID id) {
        throw new UnsupportedOperationException("Forced exception for testing purposes");
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/repository/custom/CustomGenericRequeryRepository.kt
package org.springframework.data.requery.kotlin.repository.custom

import org.springframework.data.requery.kotlin.core.RequeryOperations
import org.springframework.data.requery.kotlin.repository.support.RequeryEntityInformation
import org.springframework.data.requery.kotlin.repository.support.SimpleRequeryRepository

/**
 * org.springframework.data.requery.repository.custom.CustomGenericRequeryRepository
 *
 * @author debop
 */
class CustomGenericRequeryRepository<E: Any, ID: Any>(entityInformation: RequeryEntityInformation<E, ID>,
                                                      operations: RequeryOperations)
    : SimpleRequeryRepository<E, ID>(entityInformation, operations), CustomGenericRepository<E, ID> {

    override fun customMethod(id: ID): E {
        throw UnsupportedOperationException("Forced exception for testing purposes")
    }
}